File: backend/app/models/facility.py
import uuid
from datetime import datetime, timezone

from sqlalchemy import DateTime, ForeignKey, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class Facility(Base):
    __tablename__ = "facilities"

    id: Mapped[str] = mapped_column(
        String(36),
        primary_key=True,
        default=lambda: str(uuid.uuid4()),
    )

    owner_id: Mapped[str] = mapped_column(
        String(36),
        ForeignKey("users.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )

    name: Mapped[str] = mapped_column(
        String(200),
        nullable=False,
    )

    industry_type: Mapped[str] = mapped_column(
        String(120),
        nullable=False,
    )

    location: Mapped[str | None] = mapped_column(
        String(255),
        nullable=True,
    )

    description: Mapped[str | None] = mapped_column(
        Text,
        nullable=True,
    )

    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        default=lambda: datetime.now(timezone.utc),
        nullable=False,
    )

    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        default=lambda: datetime.now(timezone.utc),
        onupdate=lambda: datetime.now(timezone.utc),
        nullable=False,
    )

File: frontend/types/facility.ts
export interface Facility {
  id: string;
  user_id: string;
  name: string;
  facility_type: string;
  industry_type: string;
  location: string;
  country: string;
  production_unit?: string;
  annual_production?: number;
  production_unit_name?: string;
  description?: string;
  created_at: string;
  updated_at: string;
}

export interface FacilityCreateRequest {
  name: string;
  facility_type: string;
  industry_type: string;
  location: string;
  country: string;
  production_unit?: string;
  annual_production?: number;
  production_unit_name?: string;
  description?: string;
}